src/components/ColorPicker.ts

```diff
@@ -0,0 +1,89 @@
+
+import "../ui/ColorPicker.scss";
+
+export interface ColorPickerProps {
+    className?: string;
+    color: string;
+    type: PickerType;
+    mode: Mode;
+    disabled: boolean;
+    disableAlpha?: boolean;
+    style?: object;
+    displayColorPicker: boolean;
+    onChange?: Picker.ColorChangeHandler;
+    close?: () => void;
+    alertMessage?: string;
+    onChangeComplete?: Picker.ColorChangeHandler;
+    defaultColors: string[];
+}
+
+export type PickerType = "sketch" | "chrome" | "block" | "github" | "twitter" | "circle" | "hue" |
+    "alpha" | "slider" | "compact" | "material" | "swatches";
+
+export type Mode = "popover" | "input" | "inline";
+
+export class ColorPicker extends Component<ColorPickerProps, {}> {
+    private components: { [P in PickerType]: any } = {
+        sketch: Picker.SketchPicker,
+        chrome: Picker.ChromePicker,
+        block: Picker.BlockPicker,
+        github: Picker.GithubPicker,
+        twitter: Picker.TwitterPicker,
+        circle: Picker.CirclePicker,
+        hue: Picker.HuePicker,
+        alpha: Picker.AlphaPicker,
+        slider: Picker.SliderPicker,
+        compact: Picker.CompactPicker,
+        material: Picker.MaterialPicker,
+        swatches: Picker.SwatchesPicker
+    };
+
+    render() {
+        return createElement("div", {
+            className: classNames(
+                "widget-color-picker",
+                this.props.className,
+                { "widget-color-picker-disabled": this.props.disabled }
+            ),
+            style: this.props.style
+        },
+            this.props.children,
+            (this.props.displayColorPicker || this.props.mode === "inline")
+                ? this.renderPicker()
+                : null,
+            createElement(Alert, { className: "widget-color-picker-alert" }, this.props.alertMessage)
+        );
+    }
+
+    private renderPicker() {
+        const { defaultColors, disabled, mode, type } = this.props;
+        const colors = defaultColors.map((color: any) => color.color);
+
+        return createElement("div", {
+            className: classNames(
+                {
+                    "widget-color-picker-popover": (mode !== "inline" && type !== "hue" && type !== "alpha" && type !== "slider")
+                }
+            )
+        },
+            mode !== "inline"
+                ? createElement("div", { className: "widget-color-picker-cover", onClick: this.props.close })
+                : null,
+            disabled ? createElement("div", { className: "widget-color-picker-overlay" }) : null,
+            createElement(this.components[type], {
+                color: this.props.color,
+                colors: (defaultColors.length > 0 && type !== "swatches") ? colors : undefined,
+                onChange: this.props.onChange,
+                onChangeComplete: this.props.onChangeComplete,
+                presetColors: defaultColors.length > 0 ? colors : undefined,
+                triangle: "hide",
```

Comment:
why hide?

Reply:
It doesn't appear nicely

Reply:
Indeed, not needed, similar style as the date picker